ocaml/xapi/xapi_host.ml

```diff
@@ -1737,8 +1737,16 @@ let sync_tunnels ~__context ~host =
           ()
         | [(pif_ref,_)] ->
           (* this is the PIF on which we want as transport PIF; let's make it *)
+          let tunnel_rec =
+            match Db.Tunnel.get_records_where ~__context ~expr: (
+              Eq (Field "transport_PIF", Literal (Ref.string_of pif_ref))
```

Comment:
I think that you will always get an empty list here, because you are looking for the tunnel that you are about the create.

You need to get the protocol from the tunnel of the master through `master_pif_rec`:
```
let protocol =
  match Xapi_pif_helpers.get_pif_type master_pif_rec with
  | Tunnel_access tunnel -> Db.Tunnel.get_protocol ~__context ~self:tunnel
  | _ -> (* raise internal error *)
in
```
Bonus points for merge this with `get_network_of_transport_pif` above and saving some DB calls.